### ISSUE TITLE
How can you use the parallel feature in the case an external npm library logs to stdout

### ISSUE THREAD

**Issue by bhreinb** [cucumber/cucumber-js#1138]
Hi there,

Looking at the parallel feature

https://github.com/cucumber/cucumber-js/blob/master/docs/cli.md#parallel-experimental

I notice however I can't really use it, if a 3rd party library happens to be writing to the stdout. I don't know how to work around that unfortunately. Do you have any suggestion to alleviate that?

In addition does each worker thread call the cucumberjs profile file defined within the project. I assume not?

Thanks in advance.

**Comment by charlierudolph** [comment]
A workaround is to have the 3rd party library write to stderr. Otherwise we would need to change how the master / slave communicates. I'll look into using IPC as that would free `process.stdio`.

---

> In addition does each worker thread call the cucumberjs profile file defined within the project

Nope. The profile is loaded only once.

**Comment by bhreinb** [comment]
Yeah this would be really useful change to have if possible...as I'm integrating cucumber with testcafe, I plan to use the parallel feature with this integration.

**Comment by jshifflet** [comment]
I was thinking about picking this issue up.

Is the goal to use IPC between slaves and master instead of using stdout?

**Comment by charlierudolph** [comment]
I've actually already got this working with IPC in a branch. Working on getting the added test passing on windows

**Comment by charlierudolph** [comment]
Fixed in 0c76b0c1a8c02016d4a8ecb5d8ba79ace8c41582. Releasing 5.0.2 now. Please try that version and let me know if its fixed

**Comment by bhreinb** [comment]
Hi @charlierudolph 

Apologies in the delay responding to you. I have tested this across windows and posix systems and it works perfectly. You can see usage of the feature within this project https://github.com/bhreinb/SYSTAC thanks again 👍
